paasta_tools/autoscaling/autoscaling_service_lib.py

```diff
@@ -168,17 +181,56 @@ def http_metrics_provider(marathon_service_config, marathon_tasks, mesos_tasks,
     utilization = []
     for task in marathon_tasks:
         try:
-            endpoint_utilization = requests.get(
-                'http://%s:%s/%s' % (task.host, task.ports[0], endpoint),
-                headers={'User-Agent': get_user_agent()}
-            ).json()['utilization']
-            utilization.append(float(endpoint_utilization))
+            utilization.append(json_mapper(get_json_body_from_service(task.host, task.ports[0], endpoint)))
         except Exception:
             pass
```

Comment:
Yes please. When debugging this thing we are just going to wonder "why" it isn't autoscaling up or down when "clearly" something is wrong, and the debug line will say "not valid json" or "couldn't connect", etc.


Also, if there is an exception, shouldn't it count against it against it?

For example, if all workers are flooded, the utilization *should* be 100%, but in practice we won't get any output from the endpoints and we'll get... 0 or [].